cmd/cosign/cli/sign/sign.go

```diff
@@ -498,11 +498,11 @@ func keylessSigner(ctx context.Context, ko options.KeyOpts) (*SignerVerifier, er
 }
 
 func SignerFromKeyOpts(ctx context.Context, certPath string, certChainPath string, ko options.KeyOpts) (*SignerVerifier, error) {
-	if ko.Sk {
+	if ko.Sk && !ko.IssueCertificate {
 		return signerFromSecurityKey(ko.Slot)
 	}
 
-	if ko.KeyRef != "" {
+	if ko.KeyRef != "" && !ko.IssueCertificate {
 		return signerFromKeyRef(ctx, certPath, certChainPath, ko.KeyRef, ko.PassFunc)
 	}
 
```

Comment:
nit: We shouldn't have output of generating ephemeral keys if IssueCertificate is set